bentoml/_internal/models/model.py

```diff
@@ -0,0 +1,255 @@
+                self._info.dump(model_yaml)
+
+        if not self.validate():
+            logger.warning(f"Failed to create Model for {self.tag}, not saving.")
+            raise BentoMLException("Failed to save Model because it was invalid")
+
+        with model_store.register(self.tag) as model_path:
+            out_fs = fs.open_fs(model_path, create=True, writeable=True)
+            fs.mirror.mirror(self._fs, out_fs, copy_if_newer=False)
+            self._fs.close()
+            self._fs = out_fs
+
+        return SysPathModel.from_Model(self)
+
+    @classmethod
+    def from_fs(cls, item_fs: FS) -> "Model":
+        res = cls(None, item_fs)  # type: ignore
+        res._tag = res.info.tag
+
+        return res
+
+    @property
+    def path(self) -> t.Optional[str]:
+        return self.path_of("/")
+
+    def path_of(self, item: str) -> t.Optional[str]:
+        try:
+            return SysPathModel.from_Model(self).path_of(item)
+        except TypeError:
+            return None
+
+    @property
+    def info(self) -> "ModelInfo":
+        if self._info is not None:
+            return self._info
+
+        with self._fs.open(MODEL_YAML_FILENAME, "r") as model_yaml:
+            self._info = ModelInfo.from_yaml_file(model_yaml)
+            return self._info
+
+    @property
+    def creation_time(self) -> datetime:
+        return self.info["creation_time"]
+
+    def export(self, path: str):
+        out_fs = fs.open_fs(path, create=True, writeable=True)
+        fs.mirror.mirror(self._fs, out_fs, copy_if_newer=False)
+
+    def push(self):
+        pass
+
+    def load_runner(self) -> "Runner":
+        raise NotImplementedError
+
+    def validate(self):
+        return self._fs.isfile(MODEL_YAML_FILENAME)
+
+    def __str__(self):
+        return f'Bento(tag="{self.tag}", path="{self.path}")'
+
+
+class SysPathModel(Model):
+    @staticmethod
+    def from_Model(model: Model) -> "SysPathModel":
+        try:
+            model._fs.getsyspath("/")
+        except fs.errors.NoSysPath:
+            raise TypeError(f"{model} is not in the OS filesystem")
+        model.__class__ = SysPathModel
+        return model  # type: ignore
+
+    @property
+    def path(self) -> str:
+        return self.path_of("/")
+
+    def path_of(self, item: str) -> str:
+        return self._fs.getsyspath(item)
+
+    def save(
+        self, model_store: "ModelStore" = Provide[BentoMLContainer.model_store]
+    ) -> "SysPathModel":
+        if self._info is not None:
+            with self._fs.open(MODEL_YAML_FILENAME, "w") as model_yaml:
+                self._info.dump(model_yaml)
+
+        if not self.validate():
+            logger.warning(f"Failed to create Model for {self.tag}, not saving.")
+            raise BentoMLException("Failed to save Model because it was invalid")
+
+        with model_store.register(self.tag) as model_path:
+            os.rmdir(model_path)
+            shutil.move(self._fs.getsyspath("/"), model_path)
```

Comment:
This should probably be `shutil.copy`?  I ran into an issue: when building a new Bento, the `bento.create` method calls this to copy over models from the local model store to a new Bento's fs, and it actually removes the files in the local model store unexpectly

Reply:
@sauyon